test/vaadin-dialog_draggable-resizable-test.html

```diff
@@ -0,0 +1,333 @@
+<!doctype html>
+<head>
+  <meta charset="UTF-8">
+  <title>vaadin-dialog tests</title>
+  <script src="../../web-component-tester/browser.js"></script>
+  <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
+  <link rel="import" href="../../test-fixture/test-fixture.html">
+  <link rel="import" href="../../iron-test-helpers/iron-test-helpers.html">
+  <link rel="import" href="../src/vaadin-dialog.html">
+</head>
+
+<body>
+  <test-fixture id="default">
+    <template>
+      <vaadin-dialog opened>
+        <template>
+          <div>Simple dialog</div>
+        </template>
+      </vaadin-dialog>
+    </template>
+  </test-fixture>
+  
+  <test-fixture id="modeless">
+    <template>
+      <vaadin-dialog modeless draggable opened>
+        <template>
+          <div>Modeless dialog 1</div>
+        </template>
+      </vaadin-dialog>
+      <vaadin-dialog modeless draggable opened>
+        <template>
+          <div>Modeless dialog 2</div>
+        </template>
+      </vaadin-dialog>
+    </template>
+  </test-fixture>
+  
+  <test-fixture id="resizable">
+    <template>
+      <vaadin-dialog resizable opened>
+        <template>
+          <div>Resizable dialog</div>
+        </template>
+      </vaadin-dialog>
+    </template>
+  </test-fixture>
+  
+  <test-fixture id="draggable">
+    <template>
+      <vaadin-dialog draggable opened>
+        <template>
+          <div>Draggable dialog</div>
+          <button>OK</button>
+        </template>
+      </vaadin-dialog>
+    </template>
+  </test-fixture>
+  
+  <script>
+      function dispatchMouseEvent(target, type, coords, button = 0) {
+        const e = new MouseEvent(type, {
+          view: window,
+          bubbles: true,
+          cancelable: true,
+          clientX: coords.x,
+          clientY: coords.y,
+          composed: true,
+          buttons: 1,
+          button
+        });
+        target.dispatchEvent(e);
+      }
+
+      function resize(target, dx, dy, mouseButton = 0) {
+        const bounds = target.getBoundingClientRect();
+        const fromXY = {y: bounds.top + (bounds.height / 2), x: bounds.left + (bounds.width / 2)};
+        const toXY = {x: fromXY.x + dx, y: fromXY.y + dy};
+        dispatchMouseEvent(target, 'mousedown', fromXY, mouseButton);
+        dispatchMouseEvent(target, 'mousemove', fromXY, mouseButton);
+        dispatchMouseEvent(target, 'mousemove', toXY, mouseButton);
+        dispatchMouseEvent(target, 'mouseup', toXY, mouseButton);
+      }
+      
+      describe('helper methods', () => {
+        var dialog, overlayPart;
+
+        beforeEach(() => {
+          dialog = fixture('default');
+          overlayPart = dialog.$.overlay.$.overlay;
+        });
+
+        it('should set bounds correctly', () => {
+          dialog._setBounds({
+            top: 200,
+            right: 'auto',
```

Comment:
right and bottom doesn't need to be tested as they're not used

Reply:
Done.